test/models/core/CoreRegistrationValidationResultSpec.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.core
+
+import base.SpecBase
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import play.api.libs.json.{JsSuccess, Json}
+
+import java.time.LocalDate
+
+class CoreRegistrationValidationResultSpec extends AnyFreeSpec with Matchers with SpecBase {
+
+  "CoreRegistrationValidationResult" - {
+
+    "must serialise and deserialise to and from a CoreRegistrationValidationResult" - {
+
+      "with all optional fields present" in {
+
+        val coreRegistrationValidationResult: CoreRegistrationValidationResult =
+          CoreRegistrationValidationResult(
+            "IM2344433220",
+            Some("IN4747493822"),
+            "FR",
+            true,
+            Seq(Match(
+              MatchType.FixedEstablishmentQuarantinedNETP,
+              "IM0987654321",
+              Some("444444444"),
+              "DE",
+              Some(3),
+              Some(LocalDate.now().format(Match.dateFormatter)),
+              Some(LocalDate.now().format(Match.dateFormatter)),
+              None,
+              None
```

Comment:
can we add these two optional fields to this test

Reply:
Added those in